experimental/dds/tree2/src/feature-libraries/forest-summary/forestSummarizer.ts

```diff
@@ -43,24 +50,23 @@ const treeBlobKey = "ForestTree";
 export class ForestSummarizer implements Summarizable {
 	public readonly key = "Forest";
 
-	private readonly schema: StoredSchemaCollection;
-	private readonly policy: FullSchemaPolicy;
-	private readonly encodeType: TreeCompressionStrategy;
-	private readonly codec: IJsonCodec<[FieldKey, EncodedChunk][], Format>;
-	private readonly options: ICodecOptions;
+	private readonly codec: ReturnType<ForestCodec>;
 
 	public constructor(
 		private readonly forest: IEditableForest,
-		schema: StoredSchemaCollection,
-		policy: FullSchemaPolicy,
-		encodeType: TreeCompressionStrategy = TreeCompressionStrategy.Compressed,
-		options?: ICodecOptions,
+		private readonly schema: StoredSchemaCollection,
+		private readonly policy: FullSchemaPolicy,
+		private readonly encodeType: TreeCompressionStrategy,
+		fieldBatchCodec: FieldBatchCodec,
+		private readonly options: ICodecOptions = { jsonValidator: noopValidator },
 	) {
-		this.schema = schema;
-		this.policy = policy;
-		this.encodeType = encodeType;
-		this.options = options ?? { jsonValidator: noopValidator };
-		this.codec = makeForestSummarizerCodec(this.options);
+		this.codec = makeForestSummarizerCodec(
```

Comment:
Don't you need to recreate `this.codec` on schema change due to changed context?

Reply:
No. `this.codec` is a function which takes in the context and returns a json codec.

Reply:
the return type of `makeForestSummarizerCodec` is, but you're providing the context in the constructor (L66-L69 in diff). `this.codec` is typed as `ReturnType<ForestCodec>`, which is an `IJsonCodec` afaict

Reply:
I missed that the read only collection of schema passed into this can be mutated by the caller of this constructor. It doesn't even provide the API for subscribing to changes...

Good catch.

We really need better conventions (or a fixed type system) to separate values which will never change from views through which you cannot edit....

Reply:
Should be fixed now